### DISCUSSION TITLE
`sendTo` or `enqueue` action with dynamic parameter

### DISCUSSION THREAD

**Discussion by dzuncoi** [statelyai/xstate#5140] [Q&A]
Hi everyone, I'm not sure if I miss anything, but is there any way that we can call `sendTo` or `enqueue` action with dynamic parameter without adding that parameter to context first, like the 2nd example here https://stately.ai/docs/actions#send-to-action?

My use case: I want to send an event to another machine with dynamic parameter. So instead of:
```javascript
assign({ valueToSend: 'new value' })
// and then
sendTo('otherMachine', { type: 'UPDATE_VALUE', newValue: context.valueToSend })
```
do we have any more robust way to do it, e.g:
```javascript
sendTo('otherMachine', { type: 'UPDATE_VALUE', newValue: /* use new value directly here*/ })
```

**Comment by davidkpiano** [comment]
Yes:
```ts
actions: sendTo('otherMachine', ({ context }) => ({
  type: 'UPDATE_VALUE',
  newValue: context.valueToSend
});
```

**Reply by dzuncoi** [reply]
Hey @davidkpiano, then you still need an extra action to assign the `newValue` into the context before sending it to another machine, isn't it? E.g I want to send value `1000` to another machine, what I need to do is:
```javascript
actions: {
  updateOtherMachine: sendTo('otherMachine', ({ context }) => ({
    type: 'UPDATE_VALUE',
    newValue: context.valueToSend
  })
}
//...
state_a: {
  entry: [
    assign({ valueToSend: 1000 }), // <- extra step
    'updateOtherMachine'
  ]
}
```
Is there anyway that we can directly send `1000` to other machine, without assigning it to context first? Maybe sth like:
```javascript
state_a: {
  entry: [{
    type: 'updateOtherMachine',
    valueToSend: 1000
  }]
}
```
